File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue(int[] nums, int original) {
        boolean present=false;
       for(int i=0;i<nums.length;i++){
            if(original==nums[i])   present=true;
       } 
       if(!present) return original;
        HashSet<Integer> set=new HashSet<>();
       if(present){
            for(int num: nums){
                set.add(num);
            }
       }
       while(set.contains(original)){
        original*=2;
       }
       return original;
    }
}

File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.cpp
class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        unordered_set<int> search;
        for(int i=0; i<nums.size(); i++){
            search.insert(nums[i]);
        }
        while(search.count(original)){
            original*=2;
        }
        return original;
    }
};